src/modules/SignIn.js

```diff
@@ -0,0 +1,122 @@
+import { Formik } from "formik";
+import { useHistory } from "react-router-dom";
+import FormInput from '../FormInput';
+import FormButton from '../FormButton';
+
+
+function SignIn() {
+    const initialValues = {
+        email: '',
+        password: '',
+    };
+
+    let validate = (values) => {
+        let errors = {};
+        const regex = /^[^\s@]+@[^\s@]+\.[^\s@]{2,}$/i;
+        if (!values.email) {
+            errors.email = 'Email is required';
+        } else if (!regex.test(values.email)) {
+            errors.email = 'Invalid Email';
+        } else if (values.email.length > 255) {
+            errors.email = 'Email too long';
+        }
+        if (!values.password) {
+            errors.password = 'Password is required';
+        } else if (values.password.length < 8) {
+            errors.password = 'Password too short';
+        } else if (values.password.length > 255) {
+            errors.password = 'Password too long';
+        }
+        return errors;
+    };
+
+    let history = useHistory();
+
+    const submitForm = (values, formikBag) => {
+        const jsonData = {
+            email: values.email,
+            password: values.password
+        }
+
+        const requestOptions = {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify(jsonData)
+        };
+
+        fetch('http://localhost:8888/v1/user/sign-in', requestOptions)
```

Comment:
what if the user leaves the page when your request in progress?
how you think it's can be handled?

Reply:
Hmm, the request should be 'stopped' somehow?? So that the user has to log in again after returning to the page.

Reply:
ok, so if you want to implement a stop/cancel pattern then you can go with AbortController https://developer.mozilla.org/en-US/docs/Web/API/AbortController

however, you need to understand that it may not work as you think. for example, browser sent a request, the server started handling it, and then the browser cancel the request, here 2 things may happen: 

1. server may not have cancelation logic handling, so your cancellation may help before started handling the request
2. server know how to handled cancellation logic, but cancel is actually triggered after the server already persisted data in DB, so it wouldn't do any change

in summary, a cancel request is not the best for creating/updating/deleting requests, however, in some cases, it still would be a valid pattern. a more common way to handle it is to block UI and try to prevent users from leaving the page or using an optimistic UI approach.

cancel requests is very helpful when you have read requests, which can be canceled without any problems.